### PR TITLE
[Instructor training checkout] Add solutions to 10-prog.md

### PR DESCRIPTION
**Pull request by dseuss**
This adresses the python part in #178. I added solutions for the two exercises in 10-prog.md (Programming with Databases - Python).

### CODE REVIEW

File: _episodes/10-prog.md [hunk 1]
```diff
@@ -188,6 +188,44 @@ so that they are safe to use.
...
+> >
+> > random_numbers = uniform(low=10.0, high=25.0, size=100000)
+> >
+> > connection = sqlite3.connect("original.db")
+> > cursor = connection.cursor()
+> > cursor.execute("CREATE TABLE Pressure (reading float not null)")
+> > query = "INSERT INTO Pressure values (?);"
+> >
+> > for number in random_numbers:
+> >     cursor.execute(query, [number])
```
Comment (remram44) [inline _episodes/10-prog.md]: You can actually use [`executemany()`](https://docs.python.org/3.5/library/sqlite3.html#sqlite3.Cursor.executemany) here, which should be faster

**Comment by remram44** [comment]
Thank you for the contribution! I apologize for the delay while the lesson changed maintainers. I would rather the code use executemany() (with generator syntax in the second part), but otherwise those solutions look very good! Hopefull we will get this merged in soon.

File: _episodes/10-prog.md (hunk 1 above)
Comment (dseuss) [reply _episodes/10-prog.md]: I would be careful with introducing too many new things at one. `executemany` + generator/list comprehensions might be too much at once. What do you say?

Comment (remram44) [reply _episodes/10-prog.md]: This is not a programming course, so knowledge of Python is assumed. `executemany()` makes solution 1 and 3 simpler by removing the loop, and I think the generator is easily understandable for 2 even if you didn't encounter the syntax before (`cursor_backup.execute(query, (entry for entry in results if entry[0] > 20.0))`). Do you think it's really too advanced?

**Comment by dseuss** [comment]
List comprehensions are only covered in the [intermediate and advanced](
http://intermediate-and-advanced-software-carpentry.readthedocs.io/en/latest/idiomatic-python.html)
courses as far as I know. So I wouldn’t assume familiarity in a beginners
SQL workshop.


On 29. January 2018 at 15:49:15, Remi Rampin (notifications@github.com)
wrote:

*@remram44* commented on this pull request.
------------------------------

In _episodes/10-prog.md
<https://github.com/swcarpentry/sql-novice-survey/pull/183#discussion_r164451774>
:

> +>
+> > ## Solution
+> > ~~~
+> > import sqlite3
+> > # import random number generator
+> > from numpy.random import uniform
+> >
+> > random_numbers = uniform(low=10.0, high=25.0, size=100000)
+> >
+> > connection = sqlite3.connect("original.db")
+> > cursor = connection.cursor()
+> > cursor.execute("CREATE TABLE Pressure (reading float not null)")
+> > query = "INSERT INTO Pressure values (?);"
+> >
+> > for number in random_numbers:
+> >     cursor.execute(query, [number])

This is not a programming course, so knowledge of Python is assumed.
executemany() makes solution 1 and 3 simpler by removing the loop, and I
think the generator is easily understandable for 2 even if you didn't
encounter the syntax before (cursor_backup.execute(query, (entry for entry
in results if entry[0] > 20.0))). Do you think it's really too advanced?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/swcarpentry/sql-novice-survey/pull/183#discussion_r164451774>,
or mute the thread

File: _episodes/10-prog.md (hunk 1 above)
Comment (dseuss) [reply _episodes/10-prog.md]: List comprehensions are only covered in the [intermediate and advanced](
http://intermediate-and-advanced-software-carpentry.readthedocs.io/en/latest/idiomatic-python.html)
courses as far as I know. So I wouldn’t assume familiarity in a beginners
SQL workshop.

**Pull request merged**